src/targets/gpu/compile_hip_code_object.cpp

```diff
@@ -187,6 +194,10 @@ operation compile_hip_code_object(const std::string& content, hip_compile_option
         generate_args_hpp(options.virtual_inputs.empty() ? options.inputs : options.virtual_inputs);
     srcs.push_back(src_file{fs::path{"args.hpp"},
                             std::make_pair(args_hpp.data(), args_hpp.data() + args_hpp.size())});
+    if(options.global % options.local != 0 and hip_accept_non_uniform_wg())
+        options.params += " -fno-offload-uniform-block";
+    else
+        assert(options.global % options.local == 0);
```

Comment:
This will throw an assert on older rocm that doesnt support the flag. If the flag is not supported than most likely non-uniform blocks is supported by default. I think we can remove this assert.

Reply:
if flag is not supported then it would have calculated the global WG multiple of local WG. It would work with older rocms as well. 
https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/pull/1976

Reply:
>  If the flag is not supported than most likely non-uniform blocks is supported by default 

That isn't the case with 5.7. Neither the Flag nor non-uniform blocks are supported.